src/foam/nanos/pm/PM.js

```diff
@@ -128,25 +167,47 @@ foam.CLASS({
   }
   */
 
-  public PM(ClassInfo clsInfo, String name) {
-    setName(name);
+  public PM(ClassInfo clsInfo, String... name) {
+    setName(combine(name));
     setClassType(clsInfo);
+    setId(clsInfo.getId());
     init_();
   }
 
-  public PM(Class cls, String name) {
-    setName(name);
+  public PM(Class cls, String... name) {
+    setName(combine(name));
     foam.core.ClassInfoImpl clsInfo = new foam.core.ClassInfoImpl();
     clsInfo.setObjClass(cls);
     clsInfo.setId(cls.getName());
+    setId(clsInfo.getId());
     setClassType(clsInfo);
     init_();
   }
 
-  public PM(FObject fo, String name) {
+  public PM(FObject fo, String... name) {
     this(fo.getClassInfo(), name);
   }
 
+  public PM(Object... args) {
+    setId(args[0].toString());
+    String str = "";
+    for (Object obj: args){
+      if ( obj instanceof java.lang.Exception ) {
+        str += obj.toString() + ",";
```

Comment:
if exception, 
setException(obj)
setError(((Exception) obj).getMessage());

Reply:
but wait, this is on creation.  so this is just for setting id, name. 
first arg id, rest combine for name. 
test for exception on log, error